packages/next/src/compiled/react-dom/cjs/react-dom-server.edge.production.js

```diff
@@ -6042,4 +6042,6 @@ exports.renderToReadableStream = function (children, options) {
     startWork(request);
   });
 };
+
+;const setTimeoutOrImmediate = typeof setImmediate === 'function' ? setImmediate : setTimeout;
```

Comment:
The second arg of `setTimeout` is the timeout but `...args` in setImmediate`

```suggestion
const setTimeoutOrImmediate = typeof setImmediate === 'function'
  ? setImmediate
  : <TArgs extends any[]>(
    callback: (...args: TArgs) => void,
    ...args: TArgs
  ) => setTimeout(callback, 0, ...args);
```

And remove the `0` from the second param everywhere

Reply:
this is currently safe to ignore, see comment here
https://github.com/vercel/next.js/pull/68952/files#diff-9d85a2d25346a67c95d98aa8d81b4d677b90708bdc40e03488ce0586edaade7eR1655-R1664
it is however a symptom of a deeper problem with this approach. if it *wasn't* always a `setTimeout(() => ..., 0)`, then this whole string-replace approach would be invalid anyway. this whole hack relies on the assumption that a `setTimeout(..., 0)` here means "ASAP (but after microtasks)" and thus is valid to replace with a `setImmediate`. this may not always be true if react intended to actually just wait until the next task, in which case setImmediate would be incorrect

we could be defensive against possible future `setTimeout(cb, NONZERO)` calls, by doing
```
const setTimeoutOrImmediate =
  typeof setImmediate === "function"
    ? (cb, dur = 0) => (dur === 0 ? setImmediate(cb) : setTimeout(cb, 0))
    : setTimeout;
```
or perhaps throwing an error if nonzero. which maybe is worth doing? but again, it's incomplete, because it'd miss legit uses of `setTimeout(..., 0)` that are intended to run in the next task, and not just ~soon

Reply:
We're not really making a general purpose setTimeout replacement we're trying to change the implementation of scheduleWork which is elided in the src but is the only current use of setTimeout in react-dom/server so it's fair to just tailor our solution to that. I'd revert back to the in place replacement without creating a wrapping function. The real timer functions are likely heavily optimized and adding a wrapper that might not get optimized away could introduce slowness. none of the scheduled functions take any arguments so passing 0 to them in node isn't a problem.

Reply:
makes sense to me, changed it back